### DISCUSSION TITLE
is it possible to expose electrs ?

### DISCUSSION THREAD

**Discussion by zeroday21m** [Radiokot/umbrel-cloudflared#4] [Q&A] [answered]
Is it possible to make Electrs or fulcrum public using this method, so friends and family can connect their wallets to your node, in a way thats faster then tor. 
may be on apps like blue wallet or sparrow.

**Comment by Radiokot** [comment, marked as answer]
No, it won't work.

**Reply by zeroday21m** [reply]
why is that ? 
and whats the next best way to share your electrs with friends and family on clear net ?

**Reply by Radiokot** [reply]
Cloudflare Tunnel only allows HTTP(s) traffic and Electrum server protocol is TCP-based. 
I don't know how to set up a clearnet access to Electrum server without having a white IP and an open port on your server 🤷‍♂️

**Reply by zeroday21m** [reply]
Thanks 🙏

**Comment by Radiokot** [comment]
So I ended up exposing Electrum SSL endpoint through a WireGuard tunnel to a server with a white IP, on which the traffic is proxied through Nginx stream.

**Reply by zeroday21m** [reply]
fuck. sounds complicated to a newbie. :)
any easy guide to follow ?

**Reply by Radiokot** [reply]
I don't there is one specifically for this😅

You raised a really good question. In many articles I've seen something like "if you run your own node, you could share it with your friends and family members". But on practice, sharing is not so simple.

**Reply by zeroday21m** [reply]
yeah, plus now tor option is removed from blue wallet as well. 
not a single good wallet that can access tor electrs/ fulcrum. 

so only option is clearnet in the same network/wifi.

**Reply by nkud90** [reply]
> yeah, plus now tor option is removed from blue wallet as well. not a single good wallet that can access tor electrs/ fulcrum.
> 
> so only option is clearnet in the same network/wifi.

There is an option to run Tailscale on Umbrel with Electrum Server and use Tailscale on your other devices to connect to the Electrum Server. However, this is not an ideal option because sharing access with friends is not straightforward, and Tailscale requires keeping its VPN configuration enabled while accessing the Electrum Server.

Regarding making the Electrs public - personally, I don’t understand why there is a TCP option when setting up a public hostname for the tunnel in Cloudflare, but the functionality does not work as expected. If anyone has a guess, I would appreciate your opinion.
<img width="1174" alt="TCP at Cloudflare" src="https://github.com/user-attachments/assets/7d171b05-5193-415a-8fd1-d7b1347174a0" />


By the way, many thanks to @Radiokot for the app – the Mempool Explorer runs perfectly. I hope a resolution for TCP sharing will eventually be found.

**Reply by Radiokot** [reply]
@nkud90 Thanks. The only way to have TCP work over a tunnel is by having cloudflared on both the server and the client. I think WARP also counts, but haven't tried it

**Reply by nkud90** [reply]
> @nkud90 Thanks. The only way to have TCP work over a tunnel is by having cloudflared on both the server and the client. I think WARP also counts, but haven't tried it

Yes, I reached the same conclusion regarding cloudflared but decided against it because I am still looking for a setup that would be as simple as sharing a link with a friend. I appreciate your suggestion of using a VPS with a public IP, as it effectively resolves the issue. I’m keeping it as a backup option in case no other solution becomes available in the near future.

You’ve come closest to the ideal solution with your Cloudflare Tunnel app on the Umbrel App Store, as it’s user-friendly for both the host and the end user. It’s a pity that Cloudflare’s TCP policy has let us down so far.

You are absolutely right that the community currently lacks an option to both easily run their own node and share it with friends.
